x-pack/plugins/lens/public/visualizations/xy/visualization.tsx

```diff
@@ -683,13 +721,25 @@ export const getXyVisualization = ({
       domElement
     );
   },
-
+  getAddLayerButtonComponent: (props) => {
+    return (
+      <AddLayerButton
+        {...props}
+        eventAnnotationService={eventAnnotationService}
+        onAddLayerFromAnnotationGroup={async (loadedGroupInfo) => {
+          if (loadedGroupInfo.dataViewSpec) {
+            await props.addIndexPatternFromDataViewSpec(loadedGroupInfo.dataViewSpec);
+          }
+          props.addLayer(LayerTypes.ANNOTATIONS, loadedGroupInfo);
```

Comment:
You mean instead of `addIndexPatternFromDataViewSpec`?

Reply:
I mean moving this entire logic within the `onIndexPatternChange` method, inverting the flow here implemented.
When a `dataView` changes the event should be notified to the redux state manager which should then poke any component within the visualization to change the state in a consistent way.
If the annotation layer has to react to a dataView change then the reactive logic should be implemented on that method rather than on top of the Lens central state manager, to avoid any override mistake.

Basically your code does:
`User action => Annotation specific changes => Lens redux event => onIndexPatternChange`

What it should do is:
`User action => Lens redux event => onIndexPatternChange => Annotation specific changes`